src/block_cache/block_chunkcache.c

```diff
@@ -916,6 +916,52 @@ __wt_chunkcache_get(WT_SESSION_IMPL *session, WT_BLOCK *block, uint32_t objectid
     return (0);
 }
 
+/*
+ * __wt_chunkcache_free_external --
+ *     Find chunks in the chunk cache using object id, and free the chunks.
+ */
+int
+__wt_chunkcache_free_external(
+  WT_SESSION_IMPL *session, WT_BLOCK *block, uint32_t objectid, wt_off_t offset, uint32_t size)
+{
+    WT_CHUNKCACHE *chunkcache;
+    WT_CHUNKCACHE_CHUNK *chunk, *chunk_tmp;
+    WT_CHUNKCACHE_HASHID hash_id;
+    size_t already_read;
+    uint64_t bucket_id;
+    const char *object_name;
+    chunkcache = &S2C(session)->chunkcache;
+    already_read = 0;
+    object_name = NULL;
+
+    /* Only cache read-only tiered objects. */
+    if (!block->readonly)
+        return (0);
+
+    WT_RET(
+      __wt_tiered_name(session, session->dhandle, 0, WT_TIERED_NAME_SKIP_PREFIX, &object_name));
+
+    /* A block may span multiple chunks, loop until we have read all the data. */
+    while (already_read < size) {
```

Comment:
Worth renaming `already_read` to `already_removed`? We're not doing any reading here.

Reply:
Yup this is a good point - I've changed them around alongside the comment. [6d1b1da](https://github.com/wiredtiger/wiredtiger/pull/9838/commits/6d1b1da34e6d1213840588e9088bfc8f1c674881)